### ISSUE TITLE
Fuse GDN gate and beta preprocessing in chunk training

### ISSUE THREAD

**Issue by drisspg** [pytorch-labs/attention-gym#457]
## Motivation

TorchTitan Qwen3.5 training currently materializes the GDN transforms before calling `chunk_gdn`:

```python
gate = -torch.exp(A_log.float()) * F.softplus(raw_gate.float() + dt_bias)
beta = torch.sigmoid(raw_beta)
output, _ = chunk_gdn(q, k, v, gate, beta, ...)
```

Attention Gym already fuses the equivalent softplus decay and beta sigmoid into `recurrent_gdn_decode`, but the training/chunk API accepts only the transformed FP32 tensors. This leaves separate forward operations and their backward graphs around the fused GDN kernel.

## Request

Add a supported fused GDN preprocessing path for chunk training that accepts:

- raw gate logits
- raw beta logits
- `A_log`
- `dt_bias`

and computes the Qwen3.5 GDN transforms in the kernel:

- `gate = -exp(A_log) * softplus(raw_gate + dt_bias)`
- `beta = sigmoid(raw_beta)`

The backward should return gradients for the raw inputs and parameters without requiring TorchTitan to materialize the transformed gate and beta tensors externally.

This should preserve the existing transformed-input API for callers that already have gate/beta values. Coverage should include dense and packed/varlen training, fullgraph compilation, and gradient parity against the existing eager transforms.
